spark-plugins/src/main/java/co/cask/hydrator/plugin/spark/SparkUtils.java

```diff
@@ -0,0 +1,194 @@
+   */
+  public static List<String> getFeatureList(Schema inputSchema, @Nullable String featuresToInclude,
+                                            @Nullable String featuresToExclude, String predictionField) {
+    if (!Strings.isNullOrEmpty(featuresToExclude) && !Strings.isNullOrEmpty(featuresToInclude)) {
+      throw new IllegalArgumentException("Cannot specify values for both featuresToInclude and featuresToExclude. " +
+                                           "Please specify fields for one");
+    }
+    List<String> fields = new ArrayList<>();
+    List<Schema.Field> inputSchemaFields = inputSchema.getFields();
+
+    if (!Strings.isNullOrEmpty(featuresToInclude)) {
+      List<String> features = Lists.newArrayList(Splitter.on(',').trimResults().split(featuresToInclude));
+      for (String field : features) {
+        Schema.Field inputField = inputSchema.getField(field);
+        if (!field.equals(predictionField) && inputField != null) {
+          fields.add(field);
+        }
+      }
+    } else {
+      Set<String> excludeFeatures = new HashSet<>();
+      if (!Strings.isNullOrEmpty(featuresToExclude)) {
+        excludeFeatures.addAll(Lists.newArrayList(Splitter.on(',').trimResults().split(featuresToExclude)));
+      }
+      for (Schema.Field field : inputSchemaFields) {
+        String fieldName = field.getName();
+        if (!fieldName.equals(predictionField) && (excludeFeatures.isEmpty() || !excludeFeatures.contains(fieldName))) {
+          fields.add(fieldName);
+        }
+      }
+    }
+    return fields;
+  }
+
+  /**
+   * Get the feature to cardinality mapping provided by the user.
+   *
+   * @param inputSchema        schema of the received record.
+   * @param featuresToInclude  features to be used for training/prediction.
+   * @param featuresToExclude  features to be excluded when training/predicting.
+   * @param labelField         field containing the prediction values.
+   * @param cardinalityMapping feature to cardinality mapping specified for categorical features.
+   * @return categoricalFeatureInfo for categorical features.
+   */
+  public static Map<Integer, Integer> getCategoricalFeatureInfo(Schema inputSchema, @Nullable String featuresToInclude,
+                                                                @Nullable String featuresToExclude, String labelField,
+                                                                @Nullable String cardinalityMapping) {
+    List<String> featureList = getFeatureList(inputSchema, featuresToInclude, featuresToExclude, labelField);
+    Map<Integer, Integer> outputFieldMappings = new HashMap<>();
+
+    if (!Strings.isNullOrEmpty(cardinalityMapping)) {
+      return outputFieldMappings;
+    }
+    try {
+      Map<String, String> map = Splitter.on(',').trimResults().withKeyValueSeparator(":").split(cardinalityMapping);
+      for (String field : map.keySet()) {
+        String value = map.get(field);
```

Comment:
Instead of looping on the key set and call get, iterate against the entry set.

```java
for (Map.Entry<String, String> entry : map.entrySet()) {
   ...
}
```